packages/request-manager/src/controller.ts

```diff
@@ -101,7 +101,7 @@ export class TorController extends EventEmitter {
             // for MaxCircuitDirtiness seconds after carrying the last such stream.
             // (Default: 10 minutes)
             '--MaxCircuitDirtiness',
-            '1800', // 30 minutes
+            '60', // 1 minute
```

Comment:
If I understand well, this setting makes Circuits stay alive to be reused longer time.
I understand the issue you are trying to solve is that circuits will die after 1 minute and therefore if network was changed Tor might try to use circuits that are already created but in reality are not really working.

I am afraid this can fix the edge case that you described in this Pull Request but might cause other issues in more common uses.

1800 is the configuration used in WalletWasabi https://github.com/zkSNACKs/WalletWasabi/blob/master/WalletWasabi/Tor/TorSettings.cs#L102